lib/spack/spack/build_systems/autotools.py

```diff
@@ -463,52 +531,32 @@ def build_directory(self):
         """Override to provide another place to build the package"""
         return self.configure_directory
 
-    @run_before("autoreconf")
+    @spack.builder.run_before("autoreconf")
     def delete_configure_to_force_update(self):
         if self.force_autoreconf:
-            force_remove(self.configure_abs_path)
-
-    def _require_build_deps(self, pkgs, spec, err):
-        """Require `pkgs` to be direct build dependencies of `spec`. Raises a
-        RuntimeError with a helpful error messages when any dep is missing."""
-
-        build_deps = [d.name for d in spec.dependencies(deptype="build")]
-        missing_deps = [x for x in pkgs if x not in build_deps]
-
-        if not missing_deps:
-            return
-
-        # Raise an exception on missing deps.
-        msg = (
-            "{0}: missing dependencies: {1}.\n\nPlease add "
-            "the following lines to the package:\n\n".format(err, ", ".join(missing_deps))
-        )
+            fs.force_remove(self.configure_abs_path)
 
-        for dep in missing_deps:
-            msg += "    depends_on('{0}', type='build', when='@{1}')\n".format(dep, spec.version)
-
-        msg += "\nUpdate the version (when='@{0}') as needed.".format(spec.version)
-        raise RuntimeError(msg)
-
-    def autoreconf(self, spec, prefix):
+    def autoreconf(self, pkg, spec, prefix):
```

Comment:
I mentioned this earlier, but I would still like to challenge the idea of adding `pkg` to all phases. As far as I can tell, it isn't necessary, and it breaks backwards compatibility. Is there any reason we have to add this? Will any package writers actually need to use this?